checkmates/pipelines/components.py

```diff
@@ -0,0 +1,1095 @@
+        """Fits the transformer by checking if column names are present in the dataset.
+
+        Args:
+            X (pd.DataFrame): Data to check.
+            y (pd.Series, ignored): Targets.
+
+        Returns:
+            self
+        """
+        X = infer_feature_types(X)
+        self._check_input_for_columns(X)
+        return self
+
+    def transform(self, X, y=None):
+        """Transform data using fitted column selector component.
+
+        Args:
+            X (pd.DataFrame): The input training data of shape [n_samples, n_features].
+            y (pd.Series, optional): The target training data of length [n_samples].
+
+        Returns:
+            pd.DataFrame: Transformed data.
+        """
+        X = infer_feature_types(X)
+        self._check_input_for_columns(X)
+        cols = self.parameters.get("columns") or []
+        modified_cols = self._modify_columns(cols, X, y)
+        return infer_feature_types(modified_cols)
+
+
+class DropColumns(ColumnSelector):
+    """Drops specified columns in input data.
+
+    Args:
+        columns (list(string)): List of column names, used to determine which columns to drop.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Drop Columns Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+    needs_fitting = False
+
+    def _check_input_for_columns(self, X):
+        pass
+
+    def _modify_columns(self, cols, X, y=None):
+        column_intersection = list(set(cols).intersection(X.columns))
+        return X.ww.drop(column_intersection)
+
+    def transform(self, X, y=None):
+        """Transforms data X by dropping columns.
+
+        Args:
+            X (pd.DataFrame): Data to transform.
+            y (pd.Series, optional): Targets.
+
+        Returns:
+            pd.DataFrame: Transformed X.
+        """
+        return super().transform(X, y)
+
+
+class SelectColumns(ColumnSelector):
+    """Selects specified columns in input data.
+
+    Args:
+        columns (list(string)): List of column names, used to determine which columns to select. If columns are not present, they will not be selected.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Columns Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+    needs_fitting = False
+
+    def _check_input_for_columns(self, X):
+        pass
+
+    def fit(self, X, y=None):
+        """Fits the transformer by checking if column names are present in the dataset.
+
+        Args:
+            X (pd.DataFrame): Data to check.
+            y (pd.Series, optional): Targets.
+
+        Returns:
+            self
+        """
+        return self
+
+    def _modify_columns(self, cols, X, y=None):
+        column_intersection = list(
+            sorted(set(cols).intersection(X.columns), key=cols.index),
+        )
+        return X.ww[column_intersection]
+
+
+class SelectByType(Transformer):
```

Comment:
Is this being used?

Reply:
Leaving it out is causing some errors, so keeping it in for now